extmod/vfs_lfsx.c

```diff
@@ -98,6 +98,68 @@ STATIC void MP_VFS_LFSx(init_config)(MP_OBJ_VFS_LFSx * self, mp_obj_t bdev, size
     #endif
 }
 
+// canonalize path, remove ., .. and // items
+static char *trim_path(const char *path) {
+    char *token[32];  // worst size is MAX_PATH / 2
+    char *new_path = m_malloc(strlen(path) + 1);
+    const size_t items = sizeof(token) / sizeof(char *);
+    char *in;
+    bool absolute = (path[0] == '/');
+    size_t n = 0, i;
+
+    // Break the path into pieces. 
+    strcpy(new_path, path);
+    in = new_path;
+    for (n = 0; n < items; n++) {
+        token[n] = strtok(in, "/");
+        if (token[n] == NULL) {
+            break;
+        }
+        in = NULL;
```

Comment:
shouldn't this be `in = n`?

Reply:
No. in has to be a pointer; to the input string on the first call, NULL on all following ones.

Reply:
Oops, I was not familiar with strtok, should have read the manual! What a weird interface...